moped-editor/src/components/GridTable/Filters.js

```diff
@@ -481,103 +273,26 @@ const Filters = ({
     });
   }, [setSearchParams, setFilters, setIsOr]);
 
-  /**
-   * Returns an array of strings containing messages about the filters.
-   * Returns null if no problems were found.
-   * @return {[]|null}
-   */
-  const handleApplyValidation = () => {
-    let feedback = [];
-    if (filterParameters) {
-      if (Object.keys(filterParameters).length === 0) {
-        feedback.push("• No filters have been added.");
-      } else {
-        Object.keys(filterParameters).forEach((filterKey) => {
-          const { field, value, gqlOperator } = filterParameters[filterKey];
-          if (field === null) {
-            feedback.push("• One or more fields have not been selected.");
-          }
-
-          if (gqlOperator === null) {
-            feedback.push("• One or more operators have not been selected.");
-          }
-
-          if (value === null || value.trim() === "") {
-            if (gqlOperator && !gqlOperator.includes("is_null")) {
-              feedback.push("• One or more missing values.");
-            }
-          }
-          if (!checkIsValidInput(filterParameters[filterKey])) {
-            feedback.push("• One or more invalid inputs.");
-          }
-        });
-      }
-    }
-    return feedback.length > 0 ? feedback : null;
-  };
-
   /**
    * Applies the current local state and updates the parent's state
    */
   const handleApplyButtonClick = () => {
-    const searchParamsFromFilters =
-      makeSearchParamsFromFilterParameters(filterParameters);
     setSearchParams((prevSearchParams) => {
-      const jsonParamString = JSON.stringify(searchParamsFromFilters);
+      const jsonParamString = JSON.stringify(filterParameters);
 
       prevSearchParams.set(advancedSearchFilterParamName, jsonParamString);
       prevSearchParams.set(advancedSearchIsOrParamName, isOrToggleValue);
       return prevSearchParams;
     });
 
     setIsOr(isOrToggleValue);
-    setFilters(searchParamsFromFilters);
+    setFilters(filterParameters);
     handleAdvancedSearchClose();
     // Clear simple search field in UI and state since we are using advanced search
     setSearchFieldValue("");
     setSearchTerm("");
   };
 
-  /**
-   * It returns true if the operator is null
-   * @param {object} field - The field being checked
-   * @returns {boolean}
-   */
-  const isFilterNullType = (field) => {
-    return field.gqlOperator && field.gqlOperator.includes("is_null");
-  };
-
-  /**
-   * This side effect monitors the count of filters
-   * if the count of filters is zero, then resets the state
-   */
-  useEffect(() => {
```

Comment:
Making sure that an empty filter is in `filterParameters` state now happens on first render when `initialFilterParameters` is determined. Also, since the trash can icon disappears when there is only one filter, we don't have to handle the filter count dropping to 0 that way.